source_code/main_mcu/src/fido2/fido2.c

```diff
@@ -0,0 +1,493 @@
+/*!  \file     fido2.c
+*    \brief    Crypto functions related to FIDO2
+*    Created:  15/10/2019
+*    Author:   0x0ptr@pm.me
+*
+*    Code inspired/reused from Solo project.
+*/
+#include <stdint.h>
+#include <string.h>
+#include "platform_defines.h"
+#include "comms_aux_mcu_defines.h"
+#include "comms_aux_mcu.h"
+#include "cose_key.h"
+#include "cbor.h"
+#include "crypto.h"
+#include "fido2.h"
+#include "rng.h"
+#include "main.h"
+
+enum ErrorCode
+{
+    SUCCESS = 0,
+    OPERATION_DENIED = 1,
+    USER_NOT_PRESENT = 2,
+    STORAGE_EXHAUSTED = 3,
+    CRED_NOT_FOUND = 4,
+};
+
+#define UP_BIT (1 << 0) //User Present
+#define UV_BIT (1 << 2) //User Verified
+#define AT_BIT (1 << 6) //Attested Credential Data Included
+#define ED_BIT (1 << 7) //Extension Data Included
+
+
+#define MINIBLE_AAGUID ((uint8_t*)"\x6d\xb0\x42\xd0\x61\xaf\x40\x4c\xa8\x87\xe7\x2e\x09\xba\x7e\xb4")
```

Comment:
how did you get this one?

Reply:
Random GUID generator. You can keep it or you can generate one and replace this one if you want...

Reply:
are you saying we can be creative? :D

Reply:
don't know how creative you can go. From the spec:
"Additionally, each authenticator has an AAGUID, which is a 128-bit identifier indicating the type (e.g. make and model) of the authenticator. The AAGUID MUST be chosen by the manufacturer to be identical across all substantially identical authenticators made by that manufacturer, and different (with high probability) from the AAGUIDs of all other types of authenticators. The AAGUID for a given type of authenticator **SHOULD be randomly** generated to ensure this. The Relying Party MAY use the AAGUID to infer certain properties of the authenticator, such as certification level and strength of key protection, using information from other sources."